examples/sample-p5-app/src/generators/javascript.js

```diff
@@ -11,24 +11,63 @@ import {Order} from 'blockly/javascript';
 // This file has no side effects!
 export const forBlock = Object.create(null);
 
-forBlock['add_text'] = function (block, generator) {
-  const text = generator.valueToCode(block, 'TEXT', Order.NONE) || "''";
-  const color =
-    generator.valueToCode(block, 'COLOR', Order.ATOMIC) || "'#ffffff'";
-
-  const addText = generator.provideFunction_(
-      'addText',
-      `function ${generator.FUNCTION_NAME_PLACEHOLDER_}(text, color) {
-
-  // Add text to the output area.
-  const outputDiv = document.getElementById('output');
-  const textEl = document.createElement('p');
-  textEl.innerText = text;
-  textEl.style.color = color;
-  outputDiv.appendChild(textEl);
-}`
-  );
-  // Generate the function call for this block.
-  const code = `${addText}(${text}, ${color});\n`;
+forBlock['text_print'] = function(block, generator) {
+  const msg = generator.valueToCode(block, 'TEXT',
+      Order.NONE) || '\'\'';
+  return `sketch.text(${msg}, 20, 20);\n`;
+};
+forBlock['p5_setup'] = function(block, generator) {
+  const statements = generator.statementToCode(block, 'STATEMENTS');
+  const code = `sketch.setup = function() {
+${statements}};\n`;
+  return code;
+};
+
+forBlock['p5_draw'] = function(block, generator) {
+  const statements = generator.statementToCode(block, 'STATEMENTS');
+  const code = `sketch.draw = function() {
+${statements}};\n`;
+  return code;
+};
+
+forBlock['p5_canvas'] = function(block) {
+  const width = block.getFieldValue('WIDTH') || 400;
+  const height = block.getFieldValue('HEIGHT') || 400;
+  return `sketch.createCanvas(${width}, ${height});\n`;
+};
+
+forBlock['p5_background_color'] = function(block, generator) {
+  const color = generator.valueToCode(block, 'COLOR',
+      Order.ATOMIC) || `'#fff'`;
+  // TODO: Assemble JavaScript into code variable.
```

Comment:
nit: remove TODOs here and below (unless I'm missing something this looks finished)